src/main.ts

```diff
@@ -157,9 +158,16 @@ async function mergeResultBundle(
   inputPaths: string[],
   outputPath: string
 ): Promise<void> {
+  const xcodeVersion = await getXcodeVersion();
+
   const args = ['xcresulttool', 'merge']
-    .concat(inputPaths)
-    .concat(['--output-path', outputPath])
+  .concat(inputPaths)
+  .concat(['--output-path', outputPath])
+
+  if (xcodeVersion >= 16) {
```

Comment:
I would add the `--legacy` only if the Xcode is exactly 16. Later releases may not have this flag:


```
Error: This command is deprecated and will be removed in a future release, --legacy flag is required to use it.
```